model/host/host.go

```diff
@@ -2986,9 +2986,21 @@ func GetHostByIdOrTagWithTask(hostID string) (*Host, error) {
 	return &hosts[0], nil
 }
 
-// GetPaginatedRunningHosts gets running hosts with pagination and applies any
-// filters.
-func GetPaginatedRunningHosts(hostID, distroID, currentTaskID string, statuses []string, startedBy string, sortBy string, sortDir, page, limit int) ([]Host, *int, int, error) {
+// HostsFilterOptions represents the filtering arguments for fetching hosts.
+type HostsFilterOptions struct {
+	HostID        string
+	DistroID      string
+	CurrentTaskID string
+	Statuses      []string
+	StartedBy     string
+	SortBy        string
+	SortDir       int
+	Page          int
+	Limit         int
+}
+
+// GetPaginatedRunningHosts gets running hosts with pagination and applies any filters.
+func GetPaginatedRunningHosts(opts HostsFilterOptions) ([]Host, *int, int, error) {
```

Comment:
@minnakt Since you are currently refactoring this can you thread context here so that it doesn't use the global app context to perform this query? See more [here](https://jira.mongodb.org/browse/EVG-20600)